### PR TITLE
Adding -e flag to editable local references

### LINKED ISSUE #145: Poetry export does not export local editable requirements correctly
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: MacOS Catalina 10.15.7
- **Poetry version**: 1.1.4
- **Python version**: 3.9.0
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/pawelrubin/c20a783832924952dc0c0b9fa88974f0
- **Link of a [Gist](https://gist.github.com/) with the contents of your poetry.lock file**: 
https://gist.github.com/pawelrubin/19731324b4ed3b1f3d50c6634befc016
- **Link of a [Gist](https://gist.github.com/) with the contents of your example_pkg/setup.py file**: 
https://gist.github.com/pawelrubin/563bba1d591959c674c8aca81d3918ad

## Issue

Local editable packages are not exported correctly by poetry. Thus, `pip install -r requirements.txt` fails with error:
```shell
ERROR: Invalid requirement: 'example-pkg @ /Users/pawelvewd/projects/test_poetry/example_pkg' (from line 3 of requirements.txt)
Hint: It looks like a path. File 'example-pkg @ /Users/pawelvewd/projects/test_poetry/example_pkg' does not exist.
```

### Expected result
```shell
-e ./example_pkg["foo"] ; python_version >= "3.9"
```

### Current result
```shell
example-pkg @ /Users/pawelvewd/projects/test_poetry/example_pkg; python_version >= "3.9"
```

- the path should be relative IMO
- Poetry does not export `extras`
- Poetry does not export `develop = true` (editable mode)

### PR DESCRIPTION
**Pull request by fredsensibill**
Resolves #145

Specifically, it takes the [approach suggested by @bow](https://github.com/python-poetry/poetry-plugin-export/issues/145#issuecomment-1274927714).

I would love some feedback from one of the maintainers. The issue has been open for quite some time, and I'm not entirely sure if this would be a desired behaviour.

### CODE REVIEW

File: src/poetry_plugin_export/exporter.py [hunk 1]
```diff
@@ -129,7 +127,10 @@ def _export_generic_txt(
             elif is_direct_local_reference:
                 assert dependency.source_url is not None
                 dependency_uri = path_to_url(dependency.source_url)
-                line = f"{package.complete_name} @ {dependency_uri}"
+                if package.develop:
+                    line = f"-e {dependency_uri}"
```
Comment (radoering) [inline src/poetry_plugin_export/exporter.py:131]: Does `-e` accept an URI or should this be a path?

Comment (radoering) [reply src/poetry_plugin_export/exporter.py:131]: Then, it should be fine.

**Review by radoering — APPROVED** [review]

**Comment by jbsilva** [comment]
My use case depended on the old behavior, so I simply removed the `-e` with sed: `poetry export -f requirements.txt --without-hashes | sed -E 's/^-e[[:space:]]+//' > requirements.txt"`.
I hope this helps others.

**Comment by nickroberts** [comment]
It looks like this might be why [my issue](https://github.com/python-poetry/poetry-plugin-export/issues/287) showed up. Thanks for the info, @jbsilva.

**Pull request merged**